# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/RendererD3D.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -37,7 +37,7 @@
     : mDisplay(display),
       mPresentPathFastEnabled(false),
       mCapsInitialized(false),
-      mWorkaroundsInitialized(false),
+      mFeaturesInitialized(false),
       mDisjoint(false),
       mDeviceLost(false)
 {}
@@ -72,7 +72,7 @@
     return false;
 }
 
-GLenum RendererD3D::getResetStatus()
+gl::GraphicsResetStatus RendererD3D::getResetStatus()
 {
     if (!mDeviceLost)
     {
@@ -80,17 +80,17 @@
         {
             mDeviceLost = true;
             notifyDeviceLost();
-            return GL_UNKNOWN_CONTEXT_RESET_EXT;
+            return gl::GraphicsResetStatus::UnknownContextReset;
         }
-        return GL_NO_ERROR;
+        return gl::GraphicsResetStatus::NoError;
     }
 
     if (testDeviceResettable())
     {
-        return GL_NO_ERROR;
+        return gl::GraphicsResetStatus::NoError;
     }
 
-    return GL_UNKNOWN_CONTEXT_RESET_EXT;
+    return gl::GraphicsResetStatus::UnknownContextReset;
 }
 
 void RendererD3D::notifyDeviceLost()
@@ -184,6 +184,17 @@
     return clearRenderTarget(context, renderTarget, gl::ColorF(0, 0, 0, 0), 1, 0);
 }
 
+const angle::FeaturesD3D &RendererD3D::getFeatures() const
+{
+    if (!mFeaturesInitialized)
+    {
+        initializeFeatures(&mFeatures);
+        mFeaturesInitialized = true;
+    }
+
+    return mFeatures;
+}
+
 unsigned int GetBlendSampleMask(const gl::State &glState, int samples)
 {
     unsigned int mask = 0;